test/e2e/fpga/fpga.go

```diff
@@ -0,0 +1,132 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fpga
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/intel/intel-device-plugins-for-kubernetes/test/e2e/utils"
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+const (
+	pluginDeployScript  = "scripts/deploy-fpgaplugin.sh"
+	webhookDeployScript = "scripts/webhook-deploy.sh"
+	nlb0NodeResource    = "fpga.intel.com/af-d8424dc4a4a3c413f89e433683f9040b"
+	nlb0PodResource     = "fpga.intel.com/arria10.dcp1.2-nlb0"
+	nlb3PodResource     = "fpga.intel.com/arria10.dcp1.2-nlb3"
+	arria10NodeResource = "fpga.intel.com/region-69528db6eb31577a8c3668f9faa081f6"
+)
+
+func init() {
+	ginkgo.Describe("FPGA Plugin E2E tests", describe)
+}
+
+func describe() {
+	webhookDeployScriptPath, err := utils.LocateRepoFile(webhookDeployScript)
+	if err != nil {
+		framework.Failf("unable to locate %q: %v", webhookDeployScript, err)
+	}
+
+	pluginDeployScriptPath, err := utils.LocateRepoFile(pluginDeployScript)
+	if err != nil {
+		framework.Failf("unable to locate %q: %v", pluginDeployScript, err)
+	}
+
+	fmw := framework.NewDefaultFramework("fpgaplugin-e2e")
+
+	ginkgo.It("Run FPGA plugin tests", func() {
+		// Run region test case twice to ensure that device is reprogrammed at least once
+		runTestCase(fmw, webhookDeployScriptPath, pluginDeployScriptPath, "region", "orchestrated", arria10NodeResource, nlb3PodResource, "nlb3", "nlb0")
+		runTestCase(fmw, webhookDeployScriptPath, pluginDeployScriptPath, "region", "orchestrated", arria10NodeResource, nlb0PodResource, "nlb0", "nlb3")
+		// Run af test case
+		runTestCase(fmw, webhookDeployScriptPath, pluginDeployScriptPath, "af", "preprogrammed", nlb0NodeResource, nlb0PodResource, "nlb0", "nlb3")
+	})
+}
+
+func runTestCase(fmw *framework.Framework, webhookDeployScriptPath, pluginDeployScriptPath, pluginMode, webhookMode, nodeResource, podResource, cmd1, cmd2 string) {
+	ginkgo.By(fmt.Sprintf("deploying webhook in %s mode", webhookMode))
+	_, _, err := framework.RunCmd(webhookDeployScriptPath, "--mode", webhookMode, "--namespace", fmw.Namespace.Name)
+	framework.ExpectNoError(err)
+
+	waitForPod(fmw, "intel-fpga-webhook")
+
+	ginkgo.By(fmt.Sprintf("deploying FPGA plugin in %s mode", pluginMode))
+	_, _, err = framework.RunCmd(pluginDeployScriptPath, "--mode", pluginMode, "--namespace", fmw.Namespace.Name)
+	framework.ExpectNoError(err)
```

Comment:
I've not played with FPGA deployments enough to have strong opinion here but I wonder if we should:

1. deploy a "global webhook" that gets shared between the tests (AFAIK that's the goal with #358)
2. have a custom namespace for the webhook serviceaccount
3. deploy plugin using the default serviceaccount
4. the key test case is "Note that the mappings are scoped to the namespaces they were created in and they are applicable to pods created in the corresponding namespaces." isn't that possible in each custom test run where the namespace comes from gingko?

Reply:
Currently webhook and plugin mode must match. The test uses both modes, that's why I implemented it this way. Another point of doing this is that it's a cleaner approach to use ginkgo test namespace for as many components as possible because ginkgo automatically cleans it's namespace after the test.

Reply:
@bart0sh I don't like the new script and that's why I was pondering options to drop it.

Reply:
It would be a trivial change to get rid of this script when plugin kustomization works correctly, i.e. when #318 is done. So, I'd propose to merge this PR now. Otherwise we'll have to wait until #318 is done.

Reply:
OK, that's probably fine. Do you have input to #318 to ensure "kustomization works correctly"?

Reply:
@mythi Yes, I've added 2 items to the #318 last week.